impeller/entity/shaders/rrect_blur.frag

```diff
@@ -53,7 +53,15 @@ float RRectBlurX(vec2 sample_position, vec2 half_size) {
   // The calling function RRectBlur will never provide a Y sample outside
   // of that range, though, so the max(0.0) is mostly a precaution.
   float unit_space_y = space_y / frag_info.corner_radii.y;
-  float unit_space_x = sqrt(max(0.0, 1.0 - unit_space_y * unit_space_y));
+  float unit_space_x_squared = 1.0 - unit_space_y * unit_space_y;
```

Comment:
I don't think this is necessarily an improvement. Basically we're going to take all branches and discard the results of the final else sometimes.

Reply:
I think it's more complicated than that.  If 90% don't take the branch you'll end up with warps that don't diverge right?  https://www.ece.lsu.edu/koppel/gp/2020/lsli06-br-diverg.pdf

It sounds like all branches are taken if there is ever divergence in a warp.

Reply:
When is the else true and when are the first two branches true?

Reply:
If you play around with https://gist.github.com/gaaclarke/36ed9fad08f2a34a9230a61b328ea3cc the else branch is almost never hit and when it is it's like in one group once at sigma 4 and 2 groups at sigma 20.  It's roughly 10% of the time.  That's all the intuition I have right now.

Reply:
I mentioned it in the issue too, but it seems like we can potentially get rid of the sqrt in the else branch too because in practice we are just asking for the sqrt of one number for the cases I've played around with (sigma=1, sigma=4, sigma=20).   I'll have to play with it a bit more to figure out why that is.